src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1342 @@
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: TestSuite(H).run()
+        """
+        if base_ring not in _Fields:
+            raise ValueError('base ring must be a field')
+        super().__init__(base_ring, names=names, category=Sets())
+        self._embedded = None
+        if len(names) == 2:
+            self._embedded = 'affine'
+        elif len(names) == 3:
+            self._embedded = 'projective'
+
+    def coordinate_ring(self):
+        """
+        Return the coordinate ring.
+
+        OUTPUT:
+
+        The coordinate ring of the curve arrangement.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: L.coordinate_ring()
+            Multivariate Polynomial Ring in x, y over Rational Field
+        """
+        return PolynomialRing(self.base_ring(), self.variable_names())
+
+    def change_ring(self, base_ring):
+        """
+        Return curve arrangements over a different base ring.
+
+        INPUT:
+
+        - ``base_ring`` -- a ring; the new base ring.
+
+        OUTPUT:
+
+        A new :class:`AffinePlaneCurveArrangements` instance over the new
+        base ring.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = PlaneCurveArrangements(QQ)
+            sage: L.gen(0)
+            x
+            sage: L.change_ring(RR).base_ring()
+            Real Field with 53 bits of precision
+
+        TESTS::
+
+            sage: L.change_ring(QQ) is L
+            True
+        """
+        return PlaneCurveArrangements(base_ring, names=self.variable_names())
+
+    @cached_method
+    def ambient_space(self):
+        """
+        Return the ambient space.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: L.ambient_space()
+            Affine Space of dimension 2 over Rational Field
+            sage: L.<x, y, z> = ProjectivePlaneCurveArrangements(QQ)
+            sage: L.ambient_space()
+            Projective Space of dimension 2 over Rational Field
+        """
+        n = len(self._names)
+        if n == 2:
+            return AffineSpace(self.base_ring(), 2, self._names)
+        if n == 3:
+            return ProjectiveSpace(self.base_ring(), 2, self._names)
+
+    def _repr_(self):
+        """
+        Return a string representation.
+
+        OUTPUT:
+
+        A string.
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ);  L
+            Curve arrangements in Affine Space of dimension 2 over Rational Field
+        """
+        return 'Curve arrangements in {0}'.format(self.ambient_space())
+
+    def _element_constructor_(self, *args, **kwds):
+        """
+        Construct an element of ``self``.
+
+        INPUT:
+
+        - ``*args`` -- positional arguments, each defining a curve
+
+        EXAMPLES::
+
+            sage: L.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = L._element_constructor_(x, y); A
```

Comment:
```suggestion
            sage: A = L(x, y); A
```
This type of test is better here since it should go through here, and it is more likely how the code is used. (This applies to the tests below too.)